registry/registry.go

```diff
@@ -0,0 +1,360 @@
+package registry
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/Sirupsen/logrus/formatters/logstash"
+	"github.com/bugsnag/bugsnag-go"
+	"github.com/docker/distribution/configuration"
+	ctxu "github.com/docker/distribution/context"
+	"github.com/docker/distribution/health"
+	"github.com/docker/distribution/registry/handlers"
+	"github.com/docker/distribution/registry/listener"
+	"github.com/docker/distribution/uuid"
+	"github.com/docker/distribution/version"
+	gorhandlers "github.com/gorilla/handlers"
+	"github.com/spf13/cobra"
+	"github.com/yvasiyarov/gorelic"
+	"golang.org/x/net/context"
+)
+
+// Cmd is a cobra command for running the registry.
+var Cmd = &cobra.Command{
+	Use:   "registry <config>",
+	Short: "registry stores and distributes Docker images",
+	Long:  "registry stores and distributes Docker images.",
+	Run: func(cmd *cobra.Command, args []string) {
```

Comment:
`/healthz` just returns an empty json map to indicate the http service is functional. If there is an alternative that you already have, we can use that, although we may need to support /healthz for backwards compatibility.